tokio/src/runtime/metrics/histogram/h2_histogram.rs

```diff
@@ -180,14 +193,16 @@ impl LogHistogramBuilder {
 
     /// Sets the precision of this histogram directly.
     ///
+    /// The precision (meaning: the ratio `n/bucket_range(n)` for some given `n`) will be `2^-p`.
+    ///
+    /// To match the behavior of the previous log histogram implementation, use
+    /// `builder.precision_exact(0)`. This creates a histogram where each bucket is twice the size
+    /// of the previous value.
+    ///
     /// # Panics
-    /// - `p` < 2
     /// - `p` > 10
     pub fn precision_exact(mut self, p: u32) -> Self {
-        if p < 2 {
-            panic!("precision must be >= 2");
-        };
-        if p > 10 {
+        if p > MAX_PRECISION {
             panic!("precision must be <= 10");
```

Comment:
I know this is from the previous commit, but perhaps this could be an assert instead, that's how pre-conditions are mostly checked in tokio.